File: binary-search/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
 var search = function(nums, target) {
    let begin = 0;
    let last = nums.length - 1;
    let middle;
    
    while(begin <= last)
    {
        middle = Math.floor((last+begin)/2);
        if(nums[middle] < target)
        {
            begin = middle + 1;
        }
        else if(nums[middle] > target)
        {
            last = middle - 1;
        }
        else
        {
            return middle;
        }
    }
    return -1;
};

File: binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        
        int answer = Arrays.binarySearch(nums,target);
        if (answer < 0)  
            return -1; 
        else  
            return answer;
        
        /*
        
        | BINARY SEARCH USING ALGORITHM |
        
        int middle, first = 0;
        int last = nums.length - 1;
        while(first<=last)
        {
            middle = first + (last - 1)/2;
            if (nums[middle] == target)
            {
                return middle;
            }
            if (nums[middle] < target)
            {
                first = middle + 1;
            }
            else
            {
                last = middle - 1;
            }
        }
        return -1;
        */
    }
}
